from-prototype-to-production/step-by-step.md

```diff
@@ -0,0 +1,261 @@
+- `browseURL(website)`
+- Update README with new example usage.
+
+Run checks locally, commit & push.
+
+
+### Setup GitHub Pages branch
+
+At the system shell, run
+``` bash
+git checkout --orphan gh-pages
+git rm -rf .
+echo "# Hello World" > index.md
+git add index.md
+git commit -m "Initial commit"
+git push origin gh-pages
+```
+
+You should see an Hello World website at https://USERNAME.github.io/vbzdelays.
+
+Switch back to the `master` branch.
+
+
+### Continuous deployment on Travis
+
+Setup Travis CI to automatically render and deploy the website to GitHub pages upon any push to the `master` branch.
+
+See the slides for the creation of a Personal Access Token (PAT) on GitHub and definition of the secure variable `GITHUB_PAT` on Travis.
+
+Ad the following to `.travis.yaml`:
+
+``` yaml
+before_deploy:
+- R CMD INSTALL $PKG_TARBALL
+- Rscript -e "vbzdelays::render_site()"
+
+deploy:
+  provider: pages
+  skip_cleanup: true
+  github_token: $GITHUB_PAT # Set in the settings page of your repository, as a secure variable
+  local_dir: _site
+  on:
+    branch: master
+```
+
+Commit & push, check the progress on Travis, and then browse to the rendered website  https://USERNAME.github.io/vbzdelays.
+
+
+### First release
+
+We want to start using semantic versioning for the vbzdelays package to track future updates and corresponding changes.
+
+- Setup a change log as `NEWS.md` file via `usethis::use_news_md()`, adding relevant information about this first version of the package.
+- Define a new major version (1.0.0) via `usethis::use_version("major")`.
+- Commit & push.
+- Create release on GitHub as "v1.0.0", with title "vbzdelays 1.0.0" and content of the corresponding `NEWS.md` section.
+
+
+### GitFlow setup
+
+After creating a `develop` branch (on GitHub)
+
+- Pull the branch and switch to it locally (e.g. on the Git pane of RStudio)
+- _Bump_ the package revision to include .9000: `usethis::use_dev_version()`
+- Commit & push
+
+Setup branch restrictions and agile setup as GitHub project (check slides).
+
+
+### New feature 
+
+New branch for a new feature `feature/1-add-coverage-label`, created on GitHub from `develop`. Implement the feature in the local checkout of `feature/1-add-coverage-label`.
+
+- Extend `analysis.R` and `plot.R`, run tests and examples to see the outcome.
+- Add a bullet about the new feature in the development section of `NEWS.md`.
+- Commit & push.
+- Create a pull request to `develop` on GitHub, get it approved and  merged.
+
+
+### New release 
+
+We want to release the new feature(s) as a minor release with version 1.1.0
+
+- Create branch `release/v1.1.0` from `develop`.
+- Update the package version via `usethis::use_version("minor")` locally on branch `release/v1.1.0`.
+- Commit & push.
+- Create a pull request to `master` on GitHub, get it approved and merged.
+- Create release on GitHub as "v1.1.0", with title "vbzdelays 1.1.0" and content of the corresponding `NEWS.md` section.
```

Comment:
```suggestion
- Create release on GitHub as "v1.1.0-9000", with title "vbzdelays 1.1.0" and content of the corresponding `NEWS.md` section.
```

Reply:
This is the release, so there is no -9000